airflow/providers/docker/operators/docker.py

```diff
@@ -275,32 +275,40 @@ def _run_image_with_mounts(self, target_mounts, add_tmp_variable: bool) -> Optio
             working_dir=self.working_dir,
             tty=self.tty,
         )
-        lines = self.cli.attach(container=self.container['Id'], stdout=True, stderr=True, stream=True)
+        logstream = self.cli.attach(container=self.container['Id'], stdout=True, stderr=True, stream=True)
         try:
             self.cli.start(self.container['Id'])
 
-            line = ''
-            res_lines = []
-            return_value = None
-            for line in lines:
-                if hasattr(line, 'decode'):
+            log_lines = []
+            for log_chunk in logstream:
+                if hasattr(log_chunk, 'decode'):
                     # Note that lines returned can also be byte sequences so we have to handle decode here
-                    line = line.decode('utf-8')
-                line = line.strip()
-                res_lines.append(line)
-                self.log.info(line)
+                    log_chunk = log_chunk.decode('utf-8')
```

Comment:
We probably should have `error="surrogateescape"` here (something like this, I don’t remember the exact name) because there’s no guaratee the log is outputting valid strings, and those characters shouldn’t crash the entire job.

Reply:
I haven't ruin into that before.  I'll read up on it and get the change in on Monday.

Reply:
Added in https://github.com/apache/airflow/pull/21175/commits/12df34c1d94e0c9d7fed44784d594109b3a1820d

Just making sure I understand, adding the  `surrogateescape` here allows our decode to pick up and fix anything that was escaped in the initial encode, assuming they used the same `surrogateescape` there?  Is that right?